protocol/rpcconsumer/rpcconsumer_server.go

```diff
@@ -746,24 +747,28 @@ func (rpccs *RPCConsumerServer) GetInitialUnwantedProviders(directiveHeaders map
 	return unwantedProviders
 }
 
-func (rpccs *RPCConsumerServer) HandleDirectiveHeadersForMessage(chainMessage chainlib.ChainMessage, directiveHeaders map[string]string) {
-	timeoutStr, ok := directiveHeaders[common.RELAY_TIMEOUT_HEADER_NAME]
-	if ok {
-		timeout, err := time.ParseDuration(timeoutStr)
-		if err == nil {
-			// set an override timeout
-			chainMessage.TimeoutOverride(timeout)
-		}
-	}
+func (rpccs *RPCConsumerServer) getExtensionsFromDirectiveHeaders(directiveHeaders map[string]string) []string {
 	extensionsStr, ok := directiveHeaders[common.EXTENSION_OVERRIDE_HEADER_NAME]
 	if ok {
 		extensions := strings.Split(extensionsStr, ",")
 		_, extensions, _ = rpccs.chainParser.SeparateAddonsExtensions(extensions)
 		if len(extensions) == 1 && extensions[0] == "none" {
 			// none eliminates existing extensions
-			chainMessage.OverrideExtensions([]string{}, rpccs.chainParser.ExtensionsParser())
+			return []string{}
```

Comment:
an empty list doesn't prevent from the consumer to add extensions, you need to return ExtensionInfo in this function with ExtensionInfo.ExtensionsOverride=[]string{} instead of additionalExtensons when there is a "none"